torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -84,13 +83,10 @@ Value* tryConvertToType(
 
   if (value->type()->isSubtypeOf(NoneType::get()) &&
       !concrete_type->isSubtypeOf(NoneType::get())) {
-    if (concrete_type->isSubtypeOf(OptionalType::ofTensor())) {
-      // create undefined tensor when None pass to a optional[tensor] formal arg
-      value = graph.insertNode(graph.createUndefined())->output();
-    } else if (auto optional_type = concrete_type->cast<OptionalType>()) {
-      value =
-          graph.insertNode(graph.createNone(optional_type->getElementType()))
-              ->output();
+    if (concrete_type->cast<OptionalType>()) {
+      value->setType(concrete_type);
```

Comment:
It is never ok to change the type of an already created Value. What is going on here? I am pretty sure this entire if branch is just ok to drop.